build_node/build_node_builder.py

```diff
@@ -208,7 +219,15 @@ def __report_done_task(self, task, success=True, artifacts=None):
             'status': 'done' if success else 'failed',
             'artifacts': [artifact.dict() for artifact in artifacts]
         }
-        self.__call_master('build_done', **kwargs)
+        try:
+            self.__call_master('build_done', **kwargs)
```

Comment:
You should to rewrite all call of requests.some_method if you will want to pass some additional params.
You will rewrite it only once if it will be as this:

```
if endpoint == 'build_done':
  request_method = requests.post
else:
  request_method = requests.get
request_method(full_url, json=parameters, headers=headers)
```